### PR TITLE
README updates, merge eval+retrieval inference

### PR DESCRIPTION
**Pull request by aniketrege**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # [Matryoshka Representations for Adaptive Deployment](https://arxiv.org/abs/2205.13147)
 Aditya Kusupati, Gantavya Bhatt, Aniket Rege, Matthew Wallingford, Aditya Sinha, Vivek Ramanujan, William Howard-Snyder, Kaifeng Chen, Sham Kakade, Prateek Jain, Ali Farhadi
 
-This repository contains the code for training and evaluating Matryoshka Representation with ResNet-50 backbone, along with visualizations. Training code is built upon [FFCV](https://github.com/libffcv/ffcv-imagenet) modified for Matryoshka Representations Learning (MRL). The repository is organized as follows:
+This repository contains code to train, evaluate, and analyze Matryoshka Representations with a ResNet-50 backbone. The training pipeline utilizes efficient [FFCV](https://github.com/libffcv/ffcv-imagenet) dataloaders modified for Matryoshka Representation Learning (MRL). The repository is organized as follows:
```
Comment (adityakusupati) [inline README.md]: ResNet50

File: README.md [hunk 2]
```diff
@@ -85,32 +78,32 @@ export CUDA_VISIBLE_DEVICES=0,1
...
+To evaluate our models, run the following command (arguments in brackets are optional). To evaluate the Fixed Feature Baseline, pass `--rep_size <dim>` to evaluate particular size dim. This script is also able to evaluate the standard Imagenet-1K validation set (V1). To evaluate our uploaded checkpoints, please pass `--old_ckpt`. Our model checkpoints can be found [here](https://drive.google.com/drive/folders/1IEfJk4xp-sPEKvKn6eKAUzvoRV8ho2vq?usp=sharing). 
 
 ```python
+cd inference
 python pytorch_eval.py --path <final_weigth.pt> --dataset <V2/A/Sketch/R/V1> [--tta] [--mrl] [--efficient] [--rep_size <dim>] [--old_ckpt]
 ```
 
-In paper we only consider rep sizes in  `[8, 16, 32, 64, 128, 256, 512, 1024, 2048]`, therefore to evaluate on other rep. sizes change the variable `NESTING_LIST` in file `pytorch_eval.py`. 
+In the paper we only consider $rep size \in  [8, 16, 32, 64, 128, 256, 512, 1024, 2048]$. To evaluate on other rep. sizes, change the variable `NESTING_LIST` in `pytorch_eval.py`.
```
Comment (adityakusupati) [inline README.md]: You can add accuracy plots from the paper if that makes it more convincing.

File: README.md [hunk 3]
```diff
@@ -36,81 +41,90 @@ export WRITE_DIR=/your/path/here/
...
 
 
 ## [Model Analysis](model_analysis/)
-`cd model_analysis/` 
+`cd model_analysis` 
 
-Here we provide 4 jupyter notebooks which contain performance visualization such as GradCAM images (for checkpoint models), superclass performance, model cascades and oracle upper bound. Please refer to its documentation [here](model_analysis/README.md).  
+Here we provide four jupyter notebooks which contain performance visualization such as GradCAM images (for checkpoint models), superclass performance, model cascades and oracle upper bound. Please refer to detailed documentation [here](model_analysis/README.md).  
 
 ## Retrieval Performance
```
Comment (bhattg) [inline README.md]: This should be ##[Retrieval Performance](retrieval/)

File: model_analysis/README.md [hunk 4]
```diff
@@ -1,15 +1,18 @@
 # Model Analysis
-Here provide jupyter notebooks, which contains performance visualization such as GradCAM images (for checkpoint models), superclass performance, model cascades and oracle upper bound.  
+We provide Jupyter notebooks, which contain performance visualization via GradCAM images (for checkpoint models), superclass performance, model cascades and oracle upper bound.  
 
 #### [GradCAM.ipynb](GradCAM.ipynb)
+This notebook visualizes model attribution for each image. As required preprocessing, we store each image as a torch tensor, arranged class-wise. Note that the example in the notebook is not the one shown in the paper.
 
-This notebook visualizes model attribution for each image. We beforehand store each image as torch tensor, arranged class-wise, and therefore please do so before running this script. Note that the example in the notebook is not the one shown in the paper.
+**TODO @ GB** please add a comment why and how it is different from paper
```
Comment (bhattg) [inline model_analysis/README.md]: I've taken care of this in the main branch. There will be a slight merge conflict since I changed the GradCAM.ipynb and README.md.

File: model_analysis/README.md [hunk 5]
```diff
@@ -1,15 +1,18 @@
...
+This notebook evaluates our greedy scheme for model cascading, based on maximum probability thresholding. This notebook requires the softmax predictions for the model under consideration. 
 
-#### [Custom SuperClass Performance.ipynb](<./Custom SuperClass Performance.ipynb>)
-Based on wordnet heirarchy, we evaluate our MRL model on 30 randomly chosen superclasses. The code is based on [robustness package](https://github.com/MadryLab/robustness). 	  
+#### [Custom SuperClass Performance](<./Custom_SuperClass_Performance.ipynb>)
+Based on WordNet hierarchy, we evaluate our MRL model on 30 randomly chosen superclasses. The code is based on this [robustness](https://github.com/MadryLab/robustness) project. 	  
 
-#### [Oracle Upper Bound Performance.ipynb](<./Oracle Upper Bound Performance.ipynb>)
-We compute oracle performance (the maximum possible accuracy) for MRL with ideal routing. 
+**TODO @ GB:** add WordNet link?
```
Comment (bhattg) [inline model_analysis/README.md]: Added the nltk link to wordnet.

**Review by bhattg — CHANGES_REQUESTED** [review]
Hi Aniket, I've made changes on the main branch regarding the model analysis folder. You need to be careful while merging. Other than that, this is good to go.
